File: Android_lab2/app/src/main/java/com/example/android_lab2/SecondActivity.java
package com.example.android_lab2;

import androidx.appcompat.app.AppCompatActivity;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

public class SecondActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Button button = (Button) findViewById(R.id.button);


        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {


                Intent newIntent = new Intent(SecondActivity.this, MainActivity.class);
                newIntent.putExtra("message","Введите ответ");
                startActivityForResult(newIntent, 10);
            }

        });

    }
    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
        super.onActivityResult(requestCode, resultCode, data);
        Intent intent = getIntent();
        if (requestCode == 10) {
            if (requestCode == Activity.RESULT_OK) {
                EditText editText = (EditText) findViewById(R.id.editTextNumberSigned);
                TextView textView = (TextView) findViewById(R.id.textView);
                String firstNumb = textView.getText().toString();
                String secondNumb = editText.getText().toString();
                Integer Value = MyService.Sum(firstNumb, secondNumb);
                if (intent.getStringExtra("val") != null){
                    textView.setText(intent.getStringExtra("val"));
                }
            }
        }
        setResult(Activity.RESULT_OK, intent);
    }
}

File: lab1/app/src/main/java/by/bsu/lab_1/SecondActivity.kt
package by.bsu.lab_1

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import android.widget.Toast

class SecondActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_second)

        val numberMassage = intent.getStringExtra(EXTRA_NUMBER)
        val textView = findViewById<TextView>(R.id.textNumbView).apply {
            text = numberMassage
        }

        val editText = findViewById<EditText>(R.id.editTextNumb)

        var score = numberMassage.toInt()

        val button = findViewById<Button>(R.id.button)
        button.setOnClickListener{
            val massage = editText.text.toString()
            if (massage == "")
            {
                Toast.makeText(applicationContext, "Please, enter the number", Toast.LENGTH_SHORT).show()
                return@setOnClickListener
            }
            else
            {
                score += massage.toInt()
                val intent = Intent(this, MainActivity::class.java).apply {
                    putExtra(EXTRA_NUMBER, score.toString())
                }
                startActivity(intent)
            }
        }
    }

}
